packages/web/src/store/confirmer/actions.ts

```diff
@@ -0,0 +1,198 @@
+    operationId == null
+  ) {
+    throw new Error(
+      'Programming error - must pass `operationId` in confirmation options if `squashable` or `parallelizable` is also passed.'
+    )
+  }
+  if (parallelizable === true && squashable) {
+    throw new Error(
+      'Programming error - only one of `parallelizable` or `squashable` can be true in confirmation options'
+    )
+  }
+}
+
+/**
+ * Requests a confirmation of the result of an async call, dispatching success on completion or failure on timeout.
+ * Should not be dispatched via UI, but rather a Saga/service/middleware.
+ *
+ * Confirmations with the same uid chain together and form a `confirm group`.
+ * For example, if the following actions are dispatched in rapid succession:
+ *  1. Create playlist
+ *  2. Change playlist name
+ *  3. Change playlist description
+ * The calls are fired in order in such a way that each call makes use the result of the previous call's return value.
+ * Additionally, calls within a confirm group can be further categorized using
+ * an `operationId`. Using the `confirmationOptions` param object, the requester can specify that calls of a particular `operationId`
+ * may be parallelized or squashed (where only the last call is necessary to resolve)
+ * for performance gains. The requester may also specfify to only invoke the success callback of the
+ * last call of a particular `operationId` to resolve.
+ *
+ * TODO:
+ *  - consider retries/timeout backoff
+ *  - consider other mechanisms for failure other than timeout
+ *
+ * @param uid (string) A unique identifier for this confirmation (e.g. track id)
+ * @param confirmationCall (function * | async function) The confirmation call to make. Can optionally receive the previous
+ *  confirmation's result.
+ * @param successCall (function * | async function | any An optional async call to make on success.
+ *  It is invoked with the return value of the confirmation call.
+ * @param failCall (function * | ?async function | any An optional async call to make on failure.
+ * @param previousResultSelector (function) A selector to select content from the previous confirmation to pass to this one.
+ * @param timeoutMillis (number) The millisecond timeout to await confirmation for
+ * @param confirmationOptions ConfirmationOptions object - see type documentation for details.
+ */
+export const requestConfirmation = createCustomAction(
+  REQUEST_CONFIRMATION,
+  (
+    uid: string,
+    confirmationCall: any,
+    successCall: any = function* () {},
+    failCall: any = function* () {},
+    previousResultSelector: any = () => {},
+    timeoutMillis = DEFAULT_TIMEOUT,
+    confirmationOptions: ConfirmationOptions = {}
+  ) => {
+    validateConfirmationOptions(confirmationOptions)
+    return {
+      type: REQUEST_CONFIRMATION,
+      uid,
+      confirmationCall,
+      successCall,
+      failCall,
+      previousResultSelector,
+      timeoutMillis,
+      confirmationOptions
+    }
+  }
+)
+
+/* Private */
```

Comment:
wdyt of `_prefix` notation?

Reply:
i like it! right now you can't tell what's private vs not without having to scroll a bunch - will add in the prefix